src/v/cloud_storage/remote_segment_index.cc

```diff
@@ -184,6 +184,58 @@ offset_index::find_kaf_offset(kafka::offset upper_bound) {
     return res;
 }
 
+std::optional<offset_index::find_result>
+offset_index::find_timestamp(model::timestamp upper_bound) {
+    if (_initial_time == model::timestamp::missing()) {
```

Comment:
it's a bummer that -1 is used to indicate missing timestamp. That -1 carries so much additional baggage in Kafka for example by indicating something specific in time queries.

Reply:
Yeah, it's not ideal. Not much we can do about it now though. The -1 is already encoded in a released Serde version.